flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java

```diff
@@ -443,25 +443,27 @@ protected List<Class<? extends RouterLayout>> getTargetParentLayouts(
     }
 
     private void pushHistoryStateIfNeeded(NavigationEvent event, UI ui) {
+        boolean reactEnabled = ui.getInternals().getSession().getService()
+                .getDeploymentConfiguration().isReactEnabled();
+        Location currentLocation = event.getUI().getInternals()
```

Comment:
```suggestion
        Location currentLocation = ui.getInternals()
```
I hope the given UI is the same as the event UI.

Reply:
done